reconcile/templating/lib/merge_request_manager.py

```diff
@@ -136,7 +136,7 @@ def create_merge_request(self, data: MrData) -> None:
 
         output = data.data
         collections = {o.input.collection for o in output}
-        collection_hashes = {o.input.collection_hash for o in output}
+        collection_hashes = {o.input.calc_template_hash() for o in output}
```

Comment:
the hash is only used to determine keep or close mr, right? Then why not directly use all generated content and path as hash source?

Reply:
I don't have a preference, but I think @janboll might be better at answering this question.

Reply:
I think it's better to use output, there are cases that input changed but output stay same, like changed a variable but conditional render result still stay unchanged, then no need to open new mr. To make it easier to compute hash, we can restructure `TemplateOutput` to be like

```python
class TemplateResult(BaseModel):
    input
    output
```

pick `output` field to calculate hash.

Reply:
The refactoring is bigger than I expected. I need to separate it in another PR.

Reply:
Then let's merge this first